src/services/impl/WindowsClipboard.hh

```diff
@@ -0,0 +1,47 @@
+#ifndef RA_SERVICES_WINDOWS_CLIPBOARD_HH
+#define RA_SERVICES_WINDOWS_CLIPBOARD_HH
+#pragma once
+
+#include "services\IClipboard.hh"
+
+namespace ra {
+namespace services {
+namespace impl {
+
+class WindowsClipboard : public ra::services::IClipboard
+{
+public:
+    void SetText(const std::wstring& sValue) const override
+    {
+        // allocate memory to be managed by the clipboard
+        SIZE_T nSize = (sValue.length() + 1) * 2; // 16-bit characters, with null terminator
+        HGLOBAL hMem = GlobalAlloc(GMEM_MOVEABLE, nSize);
+        if (hMem)
+        {
+            LPVOID pMem = GlobalLock(hMem);
```

Comment:
Seems this should be a const pointer since the pointer shouldn't be modified. It's kinda weird but putting const in front will do just that. As if you did "void* const".